### PR TITLE
Miscellaneous appearance fixes.

### LINKED ISSUE #57078: Monkey Mind Magnification Helmet  Sprite Error
Reporting client version: 513.1542

## Round ID:
[156773](https://scrubby.melonmesa.com/round/156773)

## Reproduction:

Tg monke helmet error

### LINKED ISSUE #57087: Runtime when interacting with modular computers related to update icon
Seems that something is wrong with modular computers and update icon. They don't change their screens as well. I attached a full stack trace from the dream daemon log. Noticed this after pulling all the recent changes. Before that pull I was running code before the ID console changes. Not sure if relevant, just the only thing I remembered that was related to modular computers lately. I'm like.. 70% sure it isn't something fucky with just my byond or the local server I've been running.. but there is always the chance it's something weird I did somehow because I am dumb. 
[message.txt](https://github.com/tgstation/tgstation/files/6016921/message.txt)

## Reproduction:
Interact with a modular computer, immediately get runtimes in the log. 
Shut down the computer, notice the icon doesn't change. 
Start a program, icon also doesn't change. 
Still get tons of runtimes in log.

### LINKED ISSUE #57102: Bloody hands with amputated arms
## Reproduction:

After having my arms amputated, I noticed that I still got bloody hand overlays above my missing hands, the blood icon floating in mid air.
My arms were clean before the surgical amputation and the blood was applied by walking over a bloody tile, I was not wearing gloves or a uniform at the time.

image

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
- Fixes massive modular computer runtimes caused by giving update_icon args it cares about.
- Fixes mind magnification helmets being errors
- Fixes the bloody hands overlay being applied to people without hands
- Fixes removing accessories also temporarily removing the bloodsplatter overlay from clothes.
- Assorted other code tweaks.
(Fixes #57102, fixes #57087, fixes #57078.)

## Why It's Good For The Game
- Less bugs

## Changelog
:cl:
fix: A recent firmware update reenabled the screens on Nanotransen brand computers.
fix: Bloodsplatter no longer experiences phantom limb syndrome on people missing both of their arms.
fix: Recent budget cuts have resulted in accessories no longer having magnets strong enough to pull bloodstains directly out of clothing. Washing machine usage has jumped 50% over the last week.
fix: No one is entirely certain who injected the the design schema for neon error signs into the mind magnification helmet production line, but the affected units have been recalled and the perpetrator <i>will</i> be brought to justice.
/:cl:

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
Oh right. Fixes #57102, fixes #57087, fixes #57078.

File: code/modules/clothing/under/_under.dm [hunk 1]
```diff
@@ -121,56 +121,65 @@
...
-	if(istype(I, /obj/item/clothing/accessory))
-		var/obj/item/clothing/accessory/A = I
-		if(attached_accessory)
-			if(user)
-				to_chat(user, "<span class='warning'>[src] already has an accessory.</span>")
-			return
-		else
+	if(!istype(I, /obj/item/clothing/accessory))
+		return
+	var/obj/item/clothing/accessory/A = I
```
Comment (Mothblocks) [inline code/modules/clothing/under/_under.dm:126]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/clothing/under/_under.dm [hunk 2]
```diff
@@ -121,56 +121,65 @@
...
+	if(!attached_accessory)
+		return
 
-		if(ishuman(loc))
-			var/mob/living/carbon/human/H = loc
-			H.update_inv_w_uniform()
-			H.update_inv_wear_suit()
-			H.fan_hud_set_fandom()
+	. = TRUE
+	var/obj/item/clothing/accessory/A = attached_accessory
```
Comment (Mothblocks) [inline code/modules/clothing/under/_under.dm:168]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/clothing/under/_under.dm [hunk 3]
```diff
@@ -121,56 +121,65 @@
...
+	if(user.put_in_hands(A))
+		to_chat(user, "<span class='notice'>You detach [A] from [src].</span>")
+	else
+		to_chat(user, "<span class='notice'>You detach [A] from [src] and it falls on the floor.</span>")
+
+	update_appearance()
+	if(!ishuman(loc))
+		return
+
+	var/mob/living/carbon/human/H = loc
```
Comment (Mothblocks) [inline code/modules/clothing/under/_under.dm:179]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/modular_computers/computers/machinery/modular_computer.dm [hunk 4]
```diff
@@ -73,6 +73,11 @@
 		. += "bsod"
 		. += "broken"
 
+/// Eats the "source" arg because update_icon actually expects args now.
+/obj/machinery/modular_computer/proc/relay_icon_update(obj/item/modular_computer/processor/source, updates, updated)
```
Comment (Mothblocks) [inline code/modules/modular_computers/computers/machinery/modular_computer.dm:77]:
```suggestion
/obj/machinery/modular_computer/proc/relay_icon_update(datum/source, updates, updated)
```

I'm pretty sure it is not guaranteed that your source will be what you expect it to be in this case, anything else could update the icon.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
> Oh right. Fixes #57102, fixes #57087, fixes #57078.

Then put it in the body, GitHub only checks the PR body.

**Comment by LemonInTheDark** [comment]
> 
> 
> Oh right. Fixes #57102, fixes #57087, fixes #57078.

Put it in the pr desc so it'll auto close pls I'm begging you

**Pull request merged**